backend/api/firebase_auth.py

```
@@ -0,0 +1,64 @@
+                    username=uid,
+                    email=email,
+                    is_active=True
+                )
+                user.save()
+            
+            # Return the user and token
+            return (user, decoded_token)
+        except auth.InvalidIdTokenError:
+            raise exceptions.AuthenticationFailed('Invalid token')
+        except Exception as e:
+            raise exceptions.AuthenticationFailed(f'Authentication failed: {str(e)}')
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)